server/src/main/java/io/spine/server/entity/storage/EntityRecordWithColumns.java

```diff
@@ -21,188 +21,141 @@
+     *         it to the type {@code I}. It is a responsibility of the caller to provide the record
+     *         with the matching identifier.
      */
-    public <R> R columnValue(ColumnName columnName, ColumnMapping<R> columnMapping) {
-        checkNotNull(columnName);
-        checkNotNull(columnMapping);
-        if (!storageFields.containsKey(columnName)) {
-            throw newIllegalStateException("Column with the name `%s` was not found.",
-                                           columnName);
-        }
-        Object columnValue = storageFields.get(columnName);
-        if (columnValue == null) {
-            R result = columnMapping.ofNull()
-                                    .apply(null);
-            return result;
-        }
-        R result = columnMapping.of(columnValue.getClass())
-                                .applyTo(columnValue);
-        return result;
+    @VisibleForTesting
+    public static <I> EntityRecordWithColumns<I>
+    of(EntityRecord record, Map<ColumnName, Object> storageFields) {
+        I id = extractId(record);
+        return new EntityRecordWithColumns<>(id, record, storageFields);
     }
 
     /**
-     * Determines if there are any {@linkplain Column columns} associated with this record.
-     *
-     * <p>If returns {@code false}, the columns are not considered by the storage.
+     * Extracts the identifier from the passed record and casts it to the type {@code I}.
      *
-     * @return {@code true} if the object was constructed using
-     *  {@link #create(EntityRecord, Entity, RecordStorage)} and the entity has columns;
-     *  {@code false} otherwise
+     * <p>It is a responsibility of the caller to provide a record with the matching identifier.
      */
-    public boolean hasColumns() {
-        return hasStorageFields;
-    }
-
-    /**
-     * Determines if there is a column with the specified name among the storage fields.
-     */
-    @SuppressWarnings("BooleanMethodIsAlwaysInverted")
-    public boolean hasColumn(ColumnName name) {
-        boolean result = storageFields.containsKey(name);
-        return result;
+    @SuppressWarnings({"unchecked", "TypeParameterUnusedInFormals"})    // see the docs.
+    private static <I> I extractId(EntityRecord record) {
+        return (I) Identifier.unpack(record.getEntityId());
     }
 
     @Override
     public LifecycleFlags getLifecycleFlags() {
-        return record.getLifecycleFlags();
+        return record().getLifecycleFlags();
     }
 
     @Override
     public boolean isArchived() {
-        return record.isArchived();
+        return record().isArchived();
     }
 
     @Override
     public boolean isDeleted() {
-        return record.isDeleted();
+        return record().isDeleted();
     }
 
     @Override
     public boolean isActive() {
-        return record.isActive();
-    }
-
-    @Override
-    public boolean equals(Object o) {
```

Comment:
Don't we need `hashCode()` and `equals()` for this class? `RecordWithColumns` is a bit restrictive on `equals()`.